### PR TITLE
Empty cart#42

### LINKED ISSUE #42: User Story 29, User can empty a cart that has items
```
As a visitor or registered user
When I have items in my cart
And I visit my cart ("/cart")
And I click the link to empty my cart
Then I am returned to my cart
All items have been completely removed from my cart
The navigation bar shows 0 items in my cart
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #46

depends on #44

### PR DESCRIPTION
**Pull request by asmolentzov**
Completes User Story 29
- Adds ability to empty the cart
- Works for both visitors and registered users
- Adds `#empty` method to Cart that clears the cart
- Adds new route for delete
- All tests passing
- Fixed merge conflict

Closes #42

### CODE REVIEW

File: spec/features/cart_show_page_spec.rb [hunk 1]
```diff
@@ -59,5 +59,77 @@
...
+    within "#nav" do
+      expect(page).to have_content("Cart: 0")
+    end
+    expect(page).to_not have_content(item_1.name)
+    expect(page).to_not have_content(item_2.name)
+    expect(page).to_not have_content(item_3.name)
+    expect(page).to_not have_content(item_1.description)
+    expect(page).to_not have_content(item_2.description)
+    expect(page).to_not have_content(item_3.description)
+    expect(page).to have_content("Grand Total: $0.00")
```
Comment (Bradniedt) [inline spec/features/cart_show_page_spec.rb]: Love these tests! Well done!

**Comment by danbriechle** [comment]
Approved
